src/tests/roc_fec/test_writer_reader.cpp

```diff
@@ -2984,27 +3095,177 @@ TEST(writer_reader, reader_invalid_fec_scheme_repair_packet) {
 
         // drop other repair packets
         for (size_t i = 0; i < NumRepairPackets - NumRepairPackets / 2 - 1; ++i) {
-            packet::PacketPtr p = writer_queue.read();
+            packet::PacketPtr p;
+            UNSIGNED_LONGS_EQUAL(status::StatusOK, writer_queue.read(p));
             CHECK(p);
             CHECK((p->flags() & packet::Packet::FlagRepair) != 0);
         }
 
         // deliver more source packets
         for (size_t i = 0; i < NumSourcePackets; ++i) {
-            packet::PacketPtr p = writer_queue.read();
+            packet::PacketPtr p;
+            UNSIGNED_LONGS_EQUAL(status::StatusOK, writer_queue.read(p));
             CHECK(p);
             CHECK((p->flags() & packet::Packet::FlagRepair) == 0);
             source_queue.write(p);
         }
         UNSIGNED_LONGS_EQUAL(NumSourcePackets, source_queue.size());
 
         // reader should shut down
-        CHECK(!reader.read());
+        packet::PacketPtr pp;
+        UNSIGNED_LONGS_EQUAL(status::StatusNoData, reader.read(pp));
+        CHECK(!pp);
         CHECK(!reader.is_alive());
         UNSIGNED_LONGS_EQUAL(0, source_queue.size());
         UNSIGNED_LONGS_EQUAL(0, repair_queue.size());
     }
 }
 
+TEST(writer_reader, failed_to_read_source_packet) {
+    for (size_t n_scheme = 0; n_scheme < CodecMap::instance().num_schemes(); ++n_scheme) {
+        codec_config.scheme = CodecMap::instance().nth_scheme(n_scheme);
+
+        const status::StatusCode codes[] = {
+            status::StatusUnknown,
+        };
+
+        for (size_t n_code = 0; n_code < ROC_ARRAY_SIZE(codes); ++n_code) {
+            core::ScopedPtr<IBlockEncoder> encoder(
+                CodecMap::instance().new_encoder(codec_config, buffer_factory, arena),
+                arena);
+
+            core::ScopedPtr<IBlockDecoder> decoder(
+                CodecMap::instance().new_decoder(codec_config, buffer_factory, arena),
+                arena);
+
+            CHECK(encoder);
+            CHECK(decoder);
+
+            StatusReader source_reader(codes[n_code]);
+            packet::Queue writer_reader;
```

Comment:
`writer_reader` is a bit trippy.. So this is actually writer to which fec::Writer will pass packets.

Maybe something like `writer_output`?

Reply:
(same for 2 tests below)